bindings/java/src/async_operator.rs

```diff
@@ -576,22 +579,41 @@ fn intern_list(
     op: *mut Operator,
     executor: *const Executor,
     path: JString,
+    options: JObject,
 ) -> Result<jlong> {
     let op = unsafe { &mut *op };
     let id = request_id(env)?;
 
     let path = jstring_to_string(env, &path)?;
 
+    let recursive = if env.call_method(&options, "isRecursive", "()Z", &[])?.z()? {
```

Comment:
i had some trouble on passing the options to the async block so that is the best way i can do. please let me know if that is not suitable, i may need some help on how to pass this arround. thanks